pkg-manager/resolve-dependencies/src/index.ts

```diff
@@ -319,20 +321,87 @@ function verifyPatches (
   })
 }
 
+export async function fetchBuildFromRegistryFS (pkgName: string, pkgVer: string, pkgUrl: string, useExperimentalNpmjsFilesIndex: boolean | string[]): Promise<boolean> {
+  interface RegistryFileFields {
+    size: number
+    type: 'File'
+    path: string
+    contentType: string
+    hex: string
+    isBinary: boolean
+    linesCount: number
+  }
+
+  interface RegistryFileList {
+    files: {
+      [path: string]: RegistryFileFields
+    }
+  }
+
+  let registryUrls = ['registry.npmjs.org/', 'registry.npmjs.com/']
+  if (Array.isArray(useExperimentalNpmjsFilesIndex)) {
+    registryUrls = useExperimentalNpmjsFilesIndex
+  }
+  // check if any of the registryUrls are contained in the pkgUrl
+  const registryUrl = registryUrls.find((url) => pkgUrl.startsWith(url))
+  if (!registryUrl) {
+    globalWarn(`${pkgName}@${pkgVer} (${pkgUrl}) is not in NPMJS Files Index URL Allow List: Fallback to requiresBuild=true`)
+    return true
```

Comment:
Just throw errors. We shouldn't create a broken lockfile just because the server didn't respond correctly. Otherwise, we'll have to also support working with broken lockfile.

Reply:
The main issue with that is that at the moment, that registry endpoint sometimes 429s when installing a package with a lot of dependencies. Not sure if you have any tips for handling that, if the other registry endpoints do the same?

Reply:
Fetch should retry on errors. Maybe you didn't pass it the retry options.